src/encodings/decoding.rs

```diff
@@ -653,6 +653,7 @@ impl<> Decoder<ByteArrayType> for DeltaByteArrayDecoder<ByteArrayType> {
       data.start_from(prefix_len_decoder.get_offset()), num_values)?;
     self.suffix_decoder = Some(suffix_decoder);
     self.num_values = num_prefixes;
+    self.current_idx = 0;
```

Comment:
I am curious if I should be resetting `self.previous_value` here.

Reply:
Added `self.previous_value.clear()` as well as `self.current_idx = 0`. I think some of tests actually cover this.